homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -327,6 +342,17 @@ def __init__(
         )
 
 
+@dataclass(frozen=True)
+class WakeSettings:
+    """Settings for wake word detection."""
+
+    timeout: float | None = None
+    """Seconds of silence before detection times out."""
+
+    audio_chunks_to_buffer: int = 0
```

Comment:
Why is this necessary for the client to pass in?

Reply:
This changes depending on if the client has audio feedback after the wake word is spoken, or if the user can just immediately say a voice command.